engine/resource/pipeline.go

```diff
@@ -95,6 +96,17 @@ type (
 		Annotations map[string]string `json:"annotations,omitempty"`
 		Labels      map[string]string `json:"labels,omitempty"`
 	}
+	// DnsConfig defines Kubernetes pod dnsConfig
+	DnsConfig struct {
+		Nameservers []string           `json:"nameservers,omitempty"`
+		Searches    []string           `json:"searches,omitempty"`
+		Options     []DNSConfigOptions `json:"options,omitempty"`
+	}
+
+	DNSConfigOptions struct {
```

Comment:
Just a minimal nitpicking, this type should be named in singular as it represents a single option.

Reply:
Sorry, completely forgot about this comment :)